patches/minecraft/net/minecraft/server/management/PlayerInteractionManager.java.patch

```diff
@@ -206,30 +206,31 @@
                  return flag1;
              }
          }
-@@ -336,6 +359,7 @@
+@@ -336,6 +359,8 @@
          }
          else
          {
-+            if (net.minecraftforge.common.ForgeHooks.onItemRightClick(p_187250_1_, p_187250_4_, p_187250_3_)) return net.minecraft.util.EnumActionResult.PASS;
++            EnumActionResult eventResult = net.minecraftforge.common.ForgeHooks.onItemRightClick(p_187250_1_, p_187250_4_, p_187250_3_);
++            if(eventResult != null) return eventResult;
              int i = p_187250_3_.field_77994_a;
              int j = p_187250_3_.func_77960_j();
              ActionResult<ItemStack> actionresult = p_187250_3_.func_77957_a(p_187250_2_, p_187250_1_, p_187250_4_);
-@@ -362,6 +386,7 @@
+@@ -362,6 +387,7 @@
                  if (itemstack.field_77994_a == 0)
                  {
                      p_187250_1_.func_184611_a(p_187250_4_, (ItemStack)null);
 +                    net.minecraftforge.event.ForgeEventFactory.onPlayerDestroyItem(p_187250_1_, itemstack, p_187250_4_);
                  }
  
                  if (!p_187250_1_.func_184587_cr())
-@@ -406,13 +431,26 @@
+@@ -406,13 +432,26 @@
          }
          else
          {
 -            if (!p_187251_1_.func_70093_af() || p_187251_1_.func_184614_ca() == null && p_187251_1_.func_184592_cb() == null)
 +            net.minecraftforge.event.entity.player.PlayerInteractEvent.RightClickBlock event = net.minecraftforge.common.ForgeHooks
```

Comment:
I've already applied the pattern you suggest to all the other events that it can go on. These ones that have `Result`'s need to hold on to the event obj